### PR TITLE
refactor(GuildMember): make `_roles` property non-enumerable

### PR DESCRIPTION
**Pull request by Idris1401**
**Please describe the changes this PR makes and why it should be merged:**

I think it would be better to have `GuildMember#roles` to be defined as a "normal" property (dunno what's the right term) instead of a getter and make `GuildMember#_roles` a getter.
People are often `console.log`ing data to get available properties and `._roles` does appear in the log and `.roles` doesn't which makes people using a private property.
I was also told that this change could cause performance issue but I still wanted to make this PR to know everyone's thought about it.

*I made another PR because I fucked up the branch in the old one.*

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

**Review by SpaceEEC — CHANGES_REQUESTED** [review]
I'd prefer `_roles` to be made non-enumerable over this change here, which would require further considerations (e.g. updates for all members when a role is deleted)

**Comment by Jiralite** [comment]
> People are often `console.log`ing data to get available properties and `._roles` does appear in the log and `.roles` doesn't which makes people using a private property.

Frankly, that's their fault. Additionally, the property is prefixed with `_` which is the convention for something that is private. I do not see this as a valid reason for this change. It instead seems like a reason to cater towards individuals that do not know what they are doing or know what they are doing is wrong.

Since #9529 landed, I am pretty sure this reverts the performance improvements.

**Comment by Idris1401** [comment]
> I'd prefer `_roles` to be made non-enumerable over this change here, which would require further considerations (e.g. updates for all members when a role is deleted)

what does "non-enumerable" mean?

**Comment by SpaceEEC** [comment]
The `descriptor` parameter of [`Object.defineProperty`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty) has an `enumerable` property, setting that to `false` is what I meant.
I think we are doing this with the `client` field in the `Base` class.

**Comment by Idris1401** [comment]
> The `descriptor` parameter of [`Object.defineProperty`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty) has an `enumerable` property, setting that to `false` is what I meant. I think we are doing this with the `client` field in the `Base` class.

based on MDN, `enumerable` is default set to `false` so I didn't add it

File: packages/discord.js/src/structures/GuildMember.js [hunk 1]
```diff
@@ -56,10 +56,12 @@ class GuildMember extends Base {
 
     /**
      * The role ids of the member
+     * @name GuildMember#_roles
      * @type {Snowflake[]}
      * @private
      */
-    this._roles = [];
+    Object.defineProperty(this, '_roles', { value: [] });
```
Comment (SpaceEEC) [inline packages/discord.js/src/structures/GuildMember.js:63]: I think you also need to add `writable` there, otherwise assigning to it will throw. 😅

**Review by SpaceEEC — CHANGES_REQUESTED** [review]

**Review by vladfrangu — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**